File: app/src/main/java/com/github/kongpf8848/animation/activity/frame/BaseFrameActivity.java
package com.github.kongpf8848.animation.activity.frame;

import android.graphics.drawable.Animatable;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.os.Looper;
import android.os.MessageQueue;
import android.view.View;
import android.widget.ImageView;

import com.github.kongpf8848.animation.activity.BaseActivity;
import com.github.kongpf8848.animation.R;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by pengf on 2017/1/7.
 */

public abstract class BaseFrameActivity extends BaseActivity {

     @BindView(R.id.anim_iv)
     ImageView imageView;

    @Override
    protected void onCreateEnd(Bundle savedInstanceState){
        super.onCreateEnd(savedInstanceState);
        Looper.myQueue().addIdleHandler(new MessageQueue.IdleHandler() {
            @Override
            public boolean queueIdle() {
                startAnim();
                return false;
            }
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        stopAnim();
    }

    private void startAnim()
    {
        Drawable drawable = imageView.getDrawable();
        if (drawable instanceof Animatable) {
            ((Animatable) drawable).start();
        }
    }

    private void stopAnim(){
        Drawable drawable = imageView.getDrawable();
        if (drawable instanceof Animatable) {
            ((Animatable) drawable).stop();
        }
    }
}

File: app/src/main/java/com/github/kongpf8848/animation/activity/frame/BaseFrameActivity.kt
package com.github.kongpf8848.animation.activity.frame

import android.graphics.drawable.Animatable
import android.os.Bundle
import android.os.Looper
import android.widget.ImageView
import com.github.kongpf8848.animation.R
import com.github.kongpf8848.animation.activity.BaseActivity

abstract class BaseFrameActivity : BaseActivity() {

    var imageView: ImageView? = null
    override fun onCreateEnd(savedInstanceState: Bundle?) {
        super.onCreateEnd(savedInstanceState)
        imageView=findViewById(R.id.anim_iv)
        Looper.myQueue().addIdleHandler {
            startAnim()
            false
        }
    }


    override fun onDestroy() {
        super.onDestroy()
        stopAnim()
    }

    private fun startAnim() {
        val drawable = imageView?.drawable
        if (drawable is Animatable) {
            drawable.start()
        }
    }

    private fun stopAnim() {
        val drawable = imageView?.drawable
        if (drawable is Animatable) {
            drawable.stop()
        }
    }
}